### PR TITLE
review CRUD

### PR DESCRIPTION
**Pull request by likuny0415**
1. add support to server

### CODE REVIEW

File: reviews/reviews-schema.js [hunk 1]
```diff
@@ -3,6 +3,8 @@ import mongoose from "mongoose";
 const reviewsSchema = mongoose.Schema({
     review: String,
     imdbID: String,
```
Comment (raymondeah) [inline reviews/reviews-schema.js]: we can delete this

File: reviews/reviews-dao.js [hunk 2]
```diff
@@ -1,13 +1,29 @@
...
+export const findReviewsBySong = (songID) =>
+  reviewsModel.find({ songID: songID }).exec();
+
+export const deleteReview = async (reviewID) => {
+  const status = await reviewsModel.deleteOne({ _id: reviewID });
+  return status;
+};
+
+
+// export const updateReview = async (reviewID, reviewUpdates) => {
```
Comment (raymondeah) [inline reviews/reviews-dao.js]: we can delete this

File: reviews/reviews-dao.js [hunk 3]
```diff
@@ -1,13 +1,29 @@
...
+//     const status = await reviewsModel.updateOne(
+//         {_id: reviewID},
+//         {$set: reviewUpdates}
+//     );
+//     return status;
+//   };
+
+
+export const updateReview = (tid, tuit) =>
```
Comment (raymondeah) [inline reviews/reviews-dao.js]: could you change the naming to relate to reviews?

File: reviews/reviews-dao.js [hunk 4]
```diff
@@ -1,13 +1,29 @@
 import reviewsModel from "./reviews-model.js";
 
-export const createReview = (review) =>
-    reviewsModel.create(review)
+export const createReview = (review) => reviewsModel.create(review);
 
 export const findReviewsByMovie = (imdbID) =>
```
Comment (raymondeah) [inline reviews/reviews-dao.js:5]: we can delete this

File: reviews/reviews-controller.js [hunk 5]
```diff
@@ -1,26 +1,59 @@
...
 
 const ReviewsController = (app) => {
-    const createReview = async (req, res) => {
-        const review = req.body
-        const currentUser = req.session['currentUser']
-        review.author = currentUser._id
-        const actualReview = await dao.createReview(review)
-        res.json(actualReview)
-    }
     const findReviewsByMovie = async (req, res) => {
```
Comment (raymondeah) [inline reviews/reviews-controller.js]: we can delete this

File: reviews/reviews-controller.js [hunk 6]
```diff
@@ -1,26 +1,59 @@
...
+
+  
+
+  app.post("/api/reviews", createReview);
+  app.get("/api/song/:songID/reviews", findReviewsBySong);
+  app.post("/api/reviews/delete/:reviewID", deleteReview)
+  app.post("/api/reviews/update/:reviewID", updateReview)
+
+  
+  app.get("/api/movies/:imdbID/reviews", findReviewsByMovie);
```
Comment (raymondeah) [inline reviews/reviews-controller.js]: we can delete this

**Comment by likuny0415** [comment]
Hi Ray, I've fixed the code can you take a look at it?

**Review by raymondeah — APPROVED** [review]
awesome

**Pull request merged**